### DISCUSSION TITLE
GetById Not mapping navigation properties properly

### DISCUSSION THREAD

**Discussion by Wasim27** [AutoMapper/AutoMapper.Extensions.OData#201] [Q&A]
Trying to fix my issue with using automapper for getting a single result back.

https://stackoverflow.com/questions/77655250/fix-automapper-odata-for-getting-a-single-result-back

```
public async Task<IEnumerable<OrderDto>> GetOrdersAsync(ODataQueryOptions<OrderDto> options)
{
    return await _context.Orders.GetQueryAsync(_mapper, options);
}

public async Task<OrderDto?> GetOrderByIdMapperAsync(int id)
{
    return await _context.Orders
        .Where(x => x.Id == id)
        .ProjectTo<OrderDto>(_mapper.ConfigurationProvider)
        .SingleOrDefaultAsync();
}
```

`CreateMap<Order, OrderDto>().ForAllMembers(opt => opt.ExplicitExpansion());`

```
public class OrderDto
{
    public int Id { get; set; }
    public string? Name { get; set; }
    public string? Description { get; set; }

    public string? CategoryName { get; set; } // this is a separate entity (navigation)
    public OrderStatusDto? OrderStatus { get; set; }
}
```

GetOrdersAsync - is working, I can query for expanded Categories and without. The query is optimised depending on whether it is expanded or not.

GetOrderByIdMapperAsync - has been impacted due to the ExplicitExpansion and just throws null for my navigation properties e.g. CategoryName is null.

If this is the wrong place for this please delete :)

See stackoverflow for more information on the controllers etc. and some things I attempted based on one response I received

**Comment by BlaiseD** [comment]
See Issue #173 .  Consider using a separate configuration for this library vs. other calls.
